PousseCafeDocletTest fails when fdp is not installed

PousseCafeDocletTest: this test seems to expect the program `fdp` to be installed.
Can you consider:
- either to remove this specific test from the ones automatically executed at build time.
- or make the test detect the presence of fdp, and avoid failure if absent.

Full log:

```Drawing BC Doc graph...
java.lang.RuntimeException: Error while writing graphs
	at poussecafe.doc.GraphImagesWriter.writeImages(GraphImagesWriter.java:42)
	at poussecafe.doc.PousseCafeDoclet.writeGraphs(PousseCafeDoclet.java:186)
	at poussecafe.doc.PousseCafeDoclet.run(PousseCafeDoclet.java:85)
	at jdk.javadoc/jdk.javadoc.internal.tool.Start.parseAndExecute(Start.java:582)
	at jdk.javadoc/jdk.javadoc.internal.tool.Start.begin(Start.java:431)
	at jdk.javadoc/jdk.javadoc.internal.tool.Start.begin(Start.java:360)
	at jdk.javadoc/jdk.javadoc.internal.api.JavadocTaskImpl.call(JavadocTaskImpl.java:96)
	at poussecafe.doc.PousseCafeDocletExecutor.execute(PousseCafeDocletExecutor.java:37)
	at poussecafe.doc.PousseCafeDocletTest.whenExecutingDoclet(PousseCafeDocletTest.java:47)
	at poussecafe.doc.PousseCafeDocletTest.docletGeneratesExpectedDoc(PousseCafeDocletTest.java:22)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
Caused by: java.io.IOException: Cannot run program "fdp": error=2, No such file or directory
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1128)
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1071)
	at poussecafe.doc.GraphImageWriter.writeImage(GraphImageWriter.java:51)
	at poussecafe.doc.GraphImagesWriter.writeImages(GraphImagesWriter.java:35)
	... 31 more
Caused by: java.io.IOException: error=2, No such file or directory
	at java.base/java.lang.ProcessImpl.forkAndExec(Native Method)
	at java.base/java.lang.ProcessImpl.<init>(ProcessImpl.java:340)
	at java.base/java.lang.ProcessImpl.start(ProcessImpl.java:271)
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1107)
	... 34 more```

--- a/pousse-cafe-doc/src/test/java/poussecafe/doc/PousseCafeDocletTest.java
+++ b/pousse-cafe-doc/src/test/java/poussecafe/doc/PousseCafeDocletTest.java
@@ -2,16 +2,23 @@ package poussecafe.doc;
 
 import java.io.File;
 import java.io.IOException;
+import java.nio.charset.Charset;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.util.Optional;
+import org.apache.commons.io.IOUtils;
 import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import static java.util.Arrays.asList;
-import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import static org.junit.Assume.assumeTrue;
 
 public class PousseCafeDocletTest {
+
+    private Logger logger = LoggerFactory.getLogger(getClass());
 
     @Test
     public void docletGeneratesExpectedDoc() {
@@ -32,6 +39,8 @@ public class PousseCafeDocletTest {
                 .customDotExecutable(Optional.of("dot"))
                 .customFdpExecutable(Optional.of("fdp"))
                 .build();
+        assumeTrue(executableInstalled("dot", configuration.customDotExecutable().orElseThrow()));
+        assumeTrue(executableInstalled("fdp", configuration.customFdpExecutable().orElseThrow()));
     }
 
     private PousseCafeDocletConfiguration configuration;
@@ -54,7 +63,21 @@ public class PousseCafeDocletTest {
                     .expectedDirectory(expectedDocDirectory)
                     .build());
         } catch (IOException e) {
-            assertTrue(false);
+            fail();
+        }
+    }
+
+    private boolean executableInstalled(String type, String executable) {
+        try {
+            Process process = new ProcessBuilder(executable, "-V").start();
+            String version = IOUtils.toString(process.getErrorStream(), Charset.defaultCharset());
+            process.waitFor();
+            logger.info("Detected custom {} executable. Version: {}", type, version);
+            return process.exitValue() == 0;
+        } catch (Exception e) {
+            logger.warn(String.format("Failed to check custom %s executable: %s", type, executable), e);
+            logger.info("Consider installing graphviz package to enable this test.");
+            return false;
         }
     }
 }

